### ISSUE TITLE
[BUG] Output window widget takes 100% CPU on large output

### ISSUE THREAD

**Issue by mattgodbolt** [compiler-explorer/compiler-explorer#2696]
Follow https://godbolt.org/z/46ETdGdYr and note the page becomes very unresponsive on each update. Profiler suggests it's in `addOutputLines` (taking ~80% of the time), so I suspect our super naïve output window doesn't like 20k lines of output.

**Comment by partouf** [comment]
Oh is that what's happening.

Why is it appearing in the output and not in the assembly windows? That's a little strange compared to the other compilers.

But yes, maybe we can optimize it somehow

**Comment by RubenRBS** [comment]
I have simplified some of the code that goes thru the output, but not what Matt identified as the major bottleneck, that remains to be tackled, but this should be a good start :)

**Comment by opensdh** [comment]
Obviously we want even more to avoid large output in the title text for the "compiler has output" icon in a conformance viewer, but the [recent change](https://github.com/compiler-explorer/compiler-explorer/commit/b2308878ea5859911951c669dceb5bd9f7d1323a#diff-e6ef2b234b8da2cad706ea8b722d6d625bfe61e7fa00d3bcea305c2ce3e1889fL806) to completely elide it seems a step too far: it's extremely convenient to check how compilers diagnose simple errors to just wave the mouse over each such icon without having to reconfigure the view.

**Comment by RubenRBS** [comment]
Oh. I'll re-implement the feature asap, sorry about that! I would have not expected anyone to actually even know that the title was there

**Comment by opensdh** [comment]
Thanks&mdash;checking multiple compilers/versions makes language issue processing much nicer!

**Comment by RubenRBS** [comment]
You're right. Sorry about that! The fix should be live once I check that everything still works ok :)

**Comment by jeremy-rifkin** [comment]
Does this still affect anyone? Performance seems fine on my end when I load the url.
